pkg/external_repos/introspect.go

```diff
@@ -106,6 +107,11 @@ func IntrospectAll() (int64, []error) {
 		return 0, []error{err}
 	}
 	for i := 0; i < len(repos); i++ {
+		hasToIntrospect, reason := needsIntrospect(&repos[i])
```

Comment:
I wonder if `needsIntrospect` should be called by `Introspect` so that the behavior also applies to `IntrospectUrl`. From a user perspective, I think I would expect `introspect [url]` and `introspect-all` to have the same behavior, except `introspect-all` applies that behavior to all the repositories.

What do you think?

Reply:
I like it, and I like more combined with the previous comment; it would provide consistency with both commands, in the other hand launching for an individual url could be useful to quickly troubleshoot the reason a specific repo is not introspected without launch it for all the repos.

Reply:
oh i see @avisiedo suggested --force above, i like that more!

Reply:
Looking at the library suggested by @rverdile to use it! thanks mates!